libp2p/network/network.go

```diff
@@ -0,0 +1,219 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/libp2p/go-libp2p/core/host"
+
+	"github.com/pokt-network/pocket/libp2p/identity"
+	"github.com/pokt-network/pocket/libp2p/protocol"
+	"github.com/pokt-network/pocket/p2p/providers"
+	"github.com/pokt-network/pocket/p2p/types"
+	poktCrypto "github.com/pokt-network/pocket/shared/crypto"
+	"github.com/pokt-network/pocket/shared/modules"
+)
+
+type libp2pNetwork struct {
+	logger      *modules.Logger
+	bus         modules.Bus
+	host        host.Host
+	topic       *pubsub.Topic
+	addrBookMap types.AddrBookMap
+}
+
+var (
+	ErrNetwork = types.NewErrFactory("LibP2P network error")
```

Comment:
reduce scope if possible

Reply:
(same rationale as https://github.com/pokt-network/pocket/pull/500#discussion_r1108124963)

Reply:
Please add a `// Exported for testing purposes` comment please